snap/info_snap_yaml.go

```diff
@@ -503,9 +503,12 @@ func bindUnboundSlots(snap *Info) {
 
 // bindImplicitHooks binds all global plugs and slots to implicit hooks
 func bindImplicitHooks(snap *Info) {
-	for _, plug := range snap.toplevelPlugs {
-		for hookName, hook := range snap.Hooks {
-			if hook.Explicit {
+	for hookName, hook := range snap.Hooks {
+		if hook.Explicit {
+			continue
+		}
+		for _, plug := range snap.toplevelPlugs {
+			if len(plug.Apps) > 0 || len(plug.Hooks) > 0 {
```

Comment:
the loop changes plug.Hooks itself that is used in the condition here so something doesn't look right, this probably needs more tests

Reply:
Ah, indeed, good point!

Reply:
This is addressed now.